Mathlib/AlgebraicGeometry/Sites/BigZariski.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.AlgebraicGeometry.Pullbacks
+import Mathlib.CategoryTheory.Sites.Pretopology
+/-!
+# The big Zariski site of schemes
+
+In this file, we define the Zariski topology, as a Grothendieck topology on the
+category `Scheme.{u}`: this is `Scheme.zariskiTopology.{u}`. If `X : Scheme.{u}`,
+the Zariski topology on `Over X` can be obtained as `Scheme.zariskiTopology.over X`
+(see `CategoryTheory.Sites.Over`.).
+
+-/
+
+universe w v u
+
+open CategoryTheory
+
+namespace AlgebraicGeometry
+
+namespace Scheme
+
+/-- The Zariski pretopology on the category of schemes. -/
+def zariskiPretopology : Pretopology (Scheme.{u}) where
+  coverings Y S := ∃ (U : OpenCover.{u} Y), S = Presieve.ofArrows U.obj U.map
+  has_isos Y X f _ := ⟨openCoverOfIsIso f, (Presieve.ofArrows_pUnit _).symm⟩
+  pullbacks := by
+    rintro Y X f _ ⟨U, rfl⟩
+    exact ⟨U.pullbackCover' f, (Presieve.ofArrows_pullback _ _ _).symm⟩
+  Transitive := by
+    rintro X _ T ⟨U, rfl⟩ H
+    let V : ∀ (j : U.J), OpenCover (U.obj j) := fun j => (H (U.map j) ⟨j⟩).choose
+    exact ⟨U.bind V, Presieve.ofArrows_bind' U.obj U.map (fun i => (V i).obj)
+      (fun i => (V i).map) T (fun j => (H (U.map j) ⟨j⟩).choose_spec)⟩
+
```

Comment:
We should have this result
```
example {Y : Scheme.{u}} (U : OpenCover.{v} Y) :
  zariskiPretopology Y (Presieve.ofArrows U.obj U.map)
```
But I'm not sure if it is easy enough to fit in this relatively small PR.
Maybe add the `OpenCover.{u} Y` version and add a TODO note?

Reply:
Thanks for the suggestion. I have included the `OpenCover.{u}` version as well as the following:
```lean
lemma zariskiTopology_openCover {Y : Scheme.{u}} (U : OpenCover.{v} Y) :
    zariskiTopology Y (Sieve.generate (Presieve.ofArrows U.obj U.map))
```